orte/mca/odls/base/odls_base_frame.c

```diff
@@ -98,12 +115,80 @@ static int orte_odls_base_register(mca_base_register_flag_t flags)
     return ORTE_SUCCESS;
 }
 
+void orte_odls_base_harvest_threads(void)
+{
+    int i;
+
+    ORTE_ACQUIRE_THREAD(&orte_odls_globals.lock);
+    if (0 < orte_odls_globals.num_threads) {
+        /* stop the progress threads */
+        for (i=0; NULL != orte_odls_globals.ev_threads[i]; i++) {
+            opal_progress_thread_finalize(orte_odls_globals.ev_threads[i]);
+        }
+        free(orte_odls_globals.ev_bases);
+        orte_odls_globals.ev_bases = (opal_event_base_t**)malloc(sizeof(opal_event_base_t*));
+        /* use the default event base */
+        orte_odls_globals.ev_bases[0] = orte_event_base;
+        orte_odls_globals.num_threads = 0;
+        if (NULL != orte_odls_globals.ev_threads) {
+            opal_argv_free(orte_odls_globals.ev_threads);
+            orte_odls_globals.ev_threads = NULL;
+        }
+    }
+    ORTE_RELEASE_THREAD(&orte_odls_globals.lock);
+}
+
+void orte_odls_base_start_threads(orte_job_t *jdata)
+{
+    int i;
+    char *tmp;
+
+    ORTE_ACQUIRE_THREAD(&orte_odls_globals.lock);
+    /* only do this once */
+    if (NULL != orte_odls_globals.ev_threads) {
+        ORTE_RELEASE_THREAD(&orte_odls_globals.lock);
+        return;
+    }
+
+    /* setup the pool of worker threads */
+    orte_odls_globals.ev_threads = NULL;
+    orte_odls_globals.next_base = 0;
+    if (0 == orte_odls_globals.num_threads ||
+        (int)jdata->num_local_procs < orte_odls_globals.cutoff) {
+        orte_odls_globals.ev_bases = (opal_event_base_t**)malloc(sizeof(opal_event_base_t*));
+        /* use the default event base */
+        orte_odls_globals.ev_bases[0] = orte_event_base;
+    } else {
+        if (-1 == orte_odls_globals.num_threads) {
+            /* user didn't specify anything, so default to some fraction of
+             * the number of local procs, capping it at the max num threads
+             * parameter value. */
+            orte_odls_globals.num_threads = jdata->num_local_procs / 8;
+            if (0 == orte_odls_globals.num_threads) {
+                orte_odls_globals.num_threads = 1;
+            } else if (orte_odls_globals.max_threads < orte_odls_globals.num_threads) {
+                orte_odls_globals.num_threads = orte_odls_globals.max_threads;
+            }
+        }
+        orte_odls_globals.ev_bases =
+            (opal_event_base_t**)malloc(orte_odls_globals.num_threads * sizeof(opal_event_base_t*));
+        for (i=0; i < orte_odls_globals.num_threads; i++) {
+            asprintf(&tmp, "ORTE-ODLS-%d", i);
+            orte_odls_globals.ev_bases[i] = opal_progress_thread_init(tmp);
+            opal_argv_append_nosize(&orte_odls_globals.ev_threads, tmp);
+            free(tmp);
+        }
+    }
+    ORTE_RELEASE_THREAD(&orte_odls_globals.lock);
+}
+
 static int orte_odls_base_close(void)
 {
     int i;
     orte_proc_t *proc;
     opal_list_item_t *item;
 
+opal_output(0, "CLOSE");
```

Comment:
is this debug code that should be removed ?

Reply:
yes, thanks - i didn't see you had made a change and was wondering why it suddenly wasn't failing! 😄